model/schema.go

```diff
@@ -0,0 +1,100 @@
+package model
+
+import (
+	"encoding/json"
+	"strings"
+)
+
+// Schema is a JSON schema for a stream.
+type Schema struct {
+	// Type is the type of the schema, e.g. "object"
```

Comment:
This is a list though 🤔

Reply:
Ah yeah good point - I'm not sure the top level needs to be list (sub-parts defo can be multiple types)